.gitmodules

```diff
@@ -0,0 +1,4 @@
+[submodule "gdal-src/source"]
```

Comment:
Is this common practice to have gdal as a git submodule or would it be better to clone it when it is needed?

Reply:
It feels like cloning could be more problematic for distros and people who want to build crates without network access.

Having a separate crate only for the source code was less familiar to me, but I guess it makes sense.

Reply:
My reasoning for using a submodule here is that this makes it much easier to test these changes locally as everything is already there. It also makes it easier to checkout potential problems and pull in additional fixes as that just need to change the commit of the submodule. 
A published version of `gdal-src` would then include a bundled version of the gdal source code inside the tar file uploaded to crates.io. That would allow other to build that package without having a git setup that allows the build script to clone gdal from somewhere. It also would prevent the build script from downloading things from the internet, which might be problematic in some situations.